src/transformers/cache_utils.py

```diff
@@ -932,8 +944,8 @@ def get_seq_length(self, layer_idx: Optional[int] = 0) -> int:
             return 0
         return self.key_cache[layer_idx].shape[-2]
 
-    def get_max_length(self) -> Optional[int]:
-        """Returns the maximum sequence length of the cached states."""
+    def get_max_cache_shape(self) -> Optional[int]:
```

Comment:
is there a specific reason for the name change ?

Reply:
The only reason is to be more specific what we mean by `max_length` since I wanted to add `max_length` in sliding cache, which currently doesn't have any (returns `None`). And as comment in code explains it was done with idea that the cache object technically handles infinite amount of tokens, so no max length

But we never want to know how many tokens can cache technically handle, our checks are all about max capacity of particular cache instance

Reply:
Let's add this in the comment for example! I was not sure 🤗